addon/services/file-manager.js

```diff
@@ -1,133 +1,266 @@
 import Ember from 'ember';
 
+/**
+ * An Ember service for doing things to files.
+ * Essentially a wrapper for the Waterbutler API.
+ * http://waterbutler.readthedocs.io/
+ *
+ * @class file-manager
+ * @extends Ember.Service
+ */
+
 export default Ember.Service.extend({
     session: Ember.inject.service(),
     store: Ember.inject.service(),
 
-    // TODO: After performing a Waterbutler action, it takes a little while
-    // (maybe a minute, depending on caching options) for the OSF API to catch
-    // up and start serving the updated metadata.
-    //
-    // For actions on existing entities, this is fine, and file-manager
-    // uses info from WB to update the Ember store so everything looks right
-    // to the user, right away.
-    //
-    // For actions that create entities (addSubfolder, uploadFile, copy),
-    // this is a problem, because we need the OSF info (guid, WB links)
-    // before letting the user do anything to their new file. At the moment,
-    // after you create a file, you should go get a cup of coffee, come back,
-    // and hard reload.
-
-    ///////////////////// File-only actions /////////////////////
-
-    getContents(file) {
+    isReloading: false,
```

Comment:
Can you explain the purpose of this flag?

In general, we'd like to try to avoid putting state on services- AFAIK these are singletons, which means that different files requests made in sequence that use the service will have the same underlying shared state.

See note below for a sample of where this could be problematic.